src/organisms/Address/Address.tsx

```diff
@@ -61,6 +61,14 @@ export class Address extends Component<Props, State> {
     this.state = { editing: false, title: props.title };
   }
 
+  public componentDidUpdate(prevProps: Readonly<Props>) {
```

Comment:
This all makes sense to me! Seems like a solid way to handle if the title changes elsewhere in the app.